### DISCUSSION TITLE
Querying parameters for netbox api calls

### DISCUSSION THREAD

**Discussion by MichaelOjoko** [netbox-community/netbox#16676] [Q&A]
I am using request module to only include fields that I need from objects in the response, 
The url looks something like this: 
https://ip-for-instance/api/dcim/devices/?brief=id 
which will only return the id, url, display, and name fields of each object as a response.

lets say I now want the racks field included in the device object, using 
https://ip-for-instance/api/dcim/devices/?brief=id, racks 

Still wont include racks, is there any other query url parameters available for netbox api?

**Comment by candlerb** [comment]
Firstly, you are querying for devices. A Netbox device doesn't have an attribute called "racks" - a device can only be in a single rack. The attribute would be called "rack".

Secondly, the "brief" query parameter doesn't take a list of fields. According to the [documentation](https://netboxlabs.com/docs/netbox/en/stable/integrations/rest-api/#brief-format) it's used like this: `?brief=1`. That's it.

What you are looking for is a new feature introduced in [v4.0](https://github.com/netbox-community/netbox/releases/tag/v4.0.0), called dynamic fields, which is query parameter `fields` rather than `brief`.

```
https://$NETBOX/api/dcim/devices/?fields=id,rack
```

**Reply by MichaelOjoko** [reply]
Rack is what I meant, the question I was really wondering was is there a way to select the fields returned by an object within the url, Ive already tried the query parameter fields so it seems im not on the right version of netbox.
Thank you.

**Reply by candlerb** [reply]
You can see what version you're running on every page, at the bottom right hand corner.
